weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/traceServerClient.ts

```diff
@@ -108,3 +108,89 @@ export class TraceServerClient {
     return res;
   };
 }
+
+const MAX_CHUNK_SIZE = 10000;
+
+/**
+ * Use this function to query calls from the trace server. This function will
+ * handle chunking the results if the user requests more than `MAX_CHUNK_SIZE`
+ * calls. This is to protect the server from returning too much data at once.
+ *
+ * Note: we should probably roll this into the `callsQuery` directly, but I don't
+ * want to change the API right now to support cancelation.
+ */
+export const chunkedCallsQuery = (
+  client: TraceServerClient,
+  req: TraceCallsQueryReq,
+  onSuccess: (res: TraceCallsQueryRes) => void,
+  onError: (err: any) => void
+): {
+  cancel: () => void;
+} => {
+  let cancelled = false;
+
+  const safeOnSuccess = (res: TraceCallsQueryRes) => {
+    if (!cancelled) {
+      onSuccess(res);
+    }
+  };
+
+  const safeOnError = (err: any) => {
+    if (!cancelled) {
+      onError(err);
+    }
+  };
+
+  const fetchCalls = async () => {
+    const userRequestedLimit = req.limit ?? Infinity;
+    const userRequestedOffset = req.offset ?? 0;
+    const shouldPage = userRequestedLimit > MAX_CHUNK_SIZE;
+    if (!shouldPage) {
+      let page: TraceCallsQueryRes;
+      try {
+        page = await client.callsQuery(req);
+      } catch (err) {
+        safeOnError(err);
+        return;
+      }
+      safeOnSuccess(page);
+    } else {
+      // Do the hard work
+      const allCallResults: TraceCallSchema[] = [];
+      let effectiveLimit = Math.min(userRequestedLimit, MAX_CHUNK_SIZE);
```

Comment:
i think this effective_limit is calculated wrong, it should be Math.max(500k, userRequestedLimit)

Reply:
Hey - thanks for catching this bug. You are correct i had an issue here. However, I solved it slightly differently. Essentially, is ensure `effectiveLimit` is always equal to `MAX_CHUNK_SIZE`. 

This makes sense to me for the following reasons:
1. I had a bug where i was decrementing the effective limit. This was totally incorrect
2. Since we are in the branch where `shouldPage = userRequestedLimit > MAX_CHUNK_SIZE` is true, then `Math.min(userRequestedLimit, MAX_CHUNK_SIZE)` will always resolve to `MAX_CHUNK_SIZE`. I tested all cases: chunk size >, =, and < total rows.